### PR TITLE
Add support for getting cpu info on Windows for llama_bench

### PR DESCRIPTION
**Pull request by kylo5aby**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [X] Low
  - [ ] Medium
  - [ ] High

When run ` llama-bench -m model_name -o json` on Windows, before this change, output: 

```shell
    "build_commit": "a15ef8f8",
    "build_number": 3416,
    "cuda": false,
    "vulkan": false,
    "kompute": false,
    "metal": false,
    "sycl": false,
    "rpc": "0",
    "gpu_blas": false,
    "blas": false,
    "cpu_info": "",
    ....
```

after this change, output:
```shell
    "build_commit": "68504f09",
    "build_number": 3455,
    "cuda": false,
    "vulkan": false,
    "kompute": false,
    "metal": false,
    "sycl": false,
    "rpc": "0",
    "gpu_blas": false,
    "blas": false,
    "cpu_info": "Intel(R) Core(TM) Ultra 5 125H",
```

### CODE REVIEW

**Comment by slaren** [comment]
Looks like this is failing on Windows ARM.

**Comment by kylo5aby** [comment]
> Looks like this is failing on Windows ARM.

Updated to support ARM, PTAL

**Comment by kylo5aby** [comment]
I see the CI/CD pipeline,  `Server / server (Release) (pull_request)` is fail, but I have run CI/Server on my forked repo, in  which CI and Server are passed, and the change won't affect `llama-server`.
here is [my CI link](https://github.com/kylo5aby/llama.cpp/actions/workflows/server.yml)
image

**Comment by slaren** [comment]
The CI failure is not related to this change, it can be ignored. The code looks good but there are two issues:
- The `hKey` is leaked on success
- `RegQueryValueEx` is not guaranteed to return a NUL-terminated string, and that needs to be handled correctly to avoid buffer overflows

**Comment by kylo5aby** [comment]
Fixed. Thanks the feedback

File: examples/llama-bench/llama-bench.cpp [hunk 1]
```diff
@@ -96,6 +104,38 @@ static std::string get_cpu_info() {
...
+    RegCloseKey(hKey);
+
+    // ensure the string is null-terminated
+    if (cpu_brand_size >= sizeof(cpu_brand)) {
+        cpu_brand[sizeof(cpu_brand) - 1] = '\0';
+    } else {
+        cpu_brand[cpu_brand_size] = '\0';
+    }
+
+    id = std::string(cpu_brand);
```
Comment (slaren) [inline examples/llama-bench/llama-bench.cpp:138]:
```suggestion
    char cpu_brand[256];
    DWORD cpu_brand_size = sizeof(cpu_brand);
    if (RegQueryValueExA(hKey,
                        TEXT("ProcessorNameString"),
                        NULL,
                        NULL,
                        (LPBYTE)cpu_brand,
                        &cpu_brand_size) == ERROR_SUCCESS) {
        id.assign(cpu_brand, cpu_brand_size);
    }
    RegCloseKey(hKey);
```

**Review by slaren — APPROVED** [review]

**Pull request merged**